Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 92532b0b85bf0a4a98b299b0606fffee43446e17
Number of Error(s): 2
CMake Error at version.cmake:8:
-- Configuring incomplete, errors occurred!
CMake Error at version.cmake:8:
  Parse error.  Expected a command name, got unquoted argument with text
  <<<<<<<.
Call Stack (most recent call first):
  CMakeLists.txt:89 (include)
-- Configuring incomplete, errors occurred!
See also /mnt/disk1/home/vamosax/smoketest/PR-12472/build/CMakeFiles/CMakeOutput.log.
npm install start.
Install ECLWatch build dependencies.
sudo npm install -g jshint@2.9.4
res:/usr/bin/jshint -> /usr/lib/node_modules/jshint/bin/jshint
+ jshint@2.9.4
updated 1 package in 2.943s
npm install end.
npm test
res:
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-12472/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-12472/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm test end
Makefiles created (2019-04-25_12-50-03 11 sec )
Build it
cmd: make -j   (2019-04-25_12-50-04)
make: *** No targets specified and no makefile found.  Stop.
res: 
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-12472/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-12472/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
Number of Error(s): 2
CMake Error at version.cmake:8:
-- Configuring incomplete, errors occurred!
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-04-25_12-49-52.log

[Build error:](https://gist.github.com/bbc59a1aa4d458712d2d9f7cc87fceb0)
Build log [Build log](https://gist.github.com/HPCCSmoketest/bbc59a1aa4d458712d2d9f7cc87fceb0#file-result-19-04-25-12-49-27-log)

--- a/esp/src/eclwatch/PackageMapPartsWidget.js
+++ b/esp/src/eclwatch/PackageMapPartsWidget.js
@@ -108,16 +108,16 @@ define([
                 this.addPartsDropDown.placeAt(this.openButton.domNode, "after");
                 this.addPartsDropDown.filterForm.set("style", "width:600px;");
                 this.addPartsDropDown.filterDropDown.set("label", context.i18n.Add);
-                this.addPartsPartName = this.createLabelAndElement("PartName", this.i18n.PartName, "ValidationTextBox", this.i18n.PartName);
-                this.addPartsContent = this.createLabelAndElement("Content", this.i18n.Content, "Textarea", this.i18n.Content);
-                this.addPartsDaliIp = this.createLabelAndElement("DaliIp", this.i18n.DaliIP, "TextBox", this.i18n.DaliIP);
-                this.addPartsSourceProcess = this.createLabelAndElement("SourceProcess", this.i18n.SourceProcess, "TextBox", this.i18n.SourceProcess);
-                this.addPartsDeletePrevious = this.createLabelAndElement("DeletePrevious", this.i18n.DeletePrevious, "CheckBox", this.i18n.DeletePrevious);
-                this.addPartsAllowForeign = this.createLabelAndElement("AllowForeignFiles", this.i18n.AllowForeignFiles, "CheckBox", this.i18n.AllowForeignFiles);
-                this.addPartsPreloadAllPackages = this.createLabelAndElement("PreloadAllPackages", this.i18n.PreloadAllPackages, "CheckBox", this.i18n.PreloadAllPackages);
-                this.addPartsUpdateSuperFiles = this.createLabelAndElement("UpdateSuperFiles", this.i18n.UpdateSuperFiles, "CheckBox", this.i18n.UpdateSuperFiles);
-                this.addPartsUpdateCloneFrom = this.createLabelAndElement("UpdateCloneFrom", this.i18n.UpdateCloneFrom, "CheckBox", this.i18n.UpdateCloneFrom);
-                this.addPartsAppendCluster = this.createLabelAndElement("AppendCluster", this.i18n.AppendCluster, "CheckBox", this.i18n.AppendCluster);
+                this.addPartsPartName = this.createLabelAndElement(this.id + "PartName", this.i18n.PartName, "ValidationTextBox", this.i18n.PartName);
+                this.addPartsContent = this.createLabelAndElement(this.id + "Content", this.i18n.Content, "Textarea", this.i18n.Content);
+                this.addPartsDaliIp = this.createLabelAndElement(this.id + "DaliIp", this.i18n.DaliIP, "TextBox", this.i18n.DaliIP);
+                this.addPartsSourceProcess = this.createLabelAndElement(this.id + "SourceProcess", this.i18n.SourceProcess, "TextBox", this.i18n.SourceProcess);
+                this.addPartsDeletePrevious = this.createLabelAndElement(this.id + "DeletePrevious", this.i18n.DeletePrevious, "CheckBox", this.i18n.DeletePrevious);
+                this.addPartsAllowForeign = this.createLabelAndElement(this.id + "AllowForeignFiles", this.i18n.AllowForeignFiles, "CheckBox", this.i18n.AllowForeignFiles);
+                this.addPartsPreloadAllPackages = this.createLabelAndElement(this.id + "PreloadAllPackages", this.i18n.PreloadAllPackages, "CheckBox", this.i18n.PreloadAllPackages);
+                this.addPartsUpdateSuperFiles = this.createLabelAndElement(this.id + "UpdateSuperFiles", this.i18n.UpdateSuperFiles, "CheckBox", this.i18n.UpdateSuperFiles);
+                this.addPartsUpdateCloneFrom = this.createLabelAndElement(this.id + "UpdateCloneFrom", this.i18n.UpdateCloneFrom, "CheckBox", this.i18n.UpdateCloneFrom);
+                this.addPartsAppendCluster = this.createLabelAndElement(this.id + "AppendCluster", this.i18n.AppendCluster, "CheckBox", this.i18n.AppendCluster);
 
                 this.removeParts = new Button({
                     id: this.id + "RemoveParts",

--- a/esp/src/eclwatch/PackageMapQueryWidget.js
+++ b/esp/src/eclwatch/PackageMapQueryWidget.js
@@ -38,6 +38,7 @@ define([
     "dijit/form/Button",
     "dijit/form/DropDownButton",
     "dijit/form/Select",
+    "dijit/form/Textarea",
     "dijit/Toolbar",
     "dijit/TooltipDialog"
 ], function (declare, lang, i18n, nlsHPCC, arrayUtil, dom, domConstruct, domForm, ObjectStore, on, topic,
@@ -71,10 +72,20 @@ define([
                 this.targetSelect = registry.byId(this.id + "TargetSelect");
                 this.processSelect = registry.byId(this.id + "ProcessSelect");
                 this.processSelectFilter = registry.byId(this.id + "ProcessFilterSelect");
+                this.addPackageMapForm = registry.byId(this.id + "AddProcessMapForm");
+                this.addProcessMapId = registry.byId(this.id + "AddProcessMapId");
+                this.addPackageMapContent = registry.byId(this.id + "AddPackagemapContent");
                 this.addPackageTargetSelect = registry.byId(this.id + "AddProcessMapTargetSelect");
                 this.addPackageProcessSelect = registry.byId(this.id + "AddProcessMapProcessSelect");
                 this.addPackageProcessFilter = registry.byId(this.id + "AddProcessMapProcessFilter");
                 this.addPackageMapDialog = registry.byId(this.id + "AddProcessMapDialog");
+                this.addPackageMapContent = dom.byId("AddPackagemapContent");
+                this.addPackageMapSubmit = registry.byId(this.id + "AddProcessMapDialogSubmit");
+                this.addPackageMapCloseButton = registry.byId(this.id + "AddProcessMapDialogClose");
+                this.addPackageMapUploader = registry.byId(this.id + "AddProcessMapFileUploader");
+                this.addProcessMapDaliIp = registry.byId(this.id + "AddProcessMapDaliIP");
+                this.addProcessMapActivate = registry.byId(this.id + "AddProcessMapActivate");
+                this.addProcessMapOverWrite = registry.byId(this.id + "AddProcessMapOverWrite");
                 this.filter = registry.byId(this.id + "Filter");
             },
 
@@ -111,82 +122,41 @@ define([
                 }
             },
 
+            _onCancel: function (event) {
+                this.addPackageMapDialog.hide();
+                this.addPackageMapContent.value = "";
+            },
+
             _onAdd: function (event) {
                 this.addPackageMapDialog.show();
-
-                var context = this;
-                var addPackageMapUploader = registry.byId(this.id + "AddProcessMapFileUploader");
-                dojo.connect(addPackageMapUploader, "onComplete", this, function (e) {
-                    registry.byId(this.id + "AddProcessMapDialogSubmit").set('disabled', false);
-                    return context.addPackageMapCallback();
-                });
-                dojo.connect(addPackageMapUploader, "onBegin", this, function (e) {
-                    registry.byId(this.id + "AddProcessMapDialogSubmit").set('disabled', true);
-                    return;
-                });
-                var addPackageMapSubmitButton = registry.byId(this.id + "AddProcessMapDialogSubmit");
-                dojo.connect(addPackageMapSubmitButton, "onClick", this, function (e) {
-                    return context._onAddPackageMapSubmit();
-                });
-                var addPackageMapCloseButton = registry.byId(this.id + "AddProcessMapDialogClose");
-                dojo.connect(addPackageMapCloseButton, "onClick", this, function (e) {
-                    this.addPackageMapDialog.onCancel();
-                });
-            },
-
-            _onAddProcessMapIdKeyUp: function () {
-                this._onCheckAddProcessMapInput();
-            },
-
-            _onCheckAddProcessMapInput: function () {
-                var id = registry.byId(this.id + "AddProcessMapId").get('value');
-                var files = registry.byId(this.id + "AddProcessMapFileUploader").getFileList();
-                if (files.length > 1) {
-                    alert(this.i18n.Only1PackageFileAllowed);
-                    return;
-                }
-                var fileName = '';
-                if (files.length > 0)
-                    fileName = files[0].name;
-                if ((fileName !== '') && (id === '')) {
-                    registry.byId(this.id + "AddProcessMapId").set('value', fileName);
-                    registry.byId(this.id + "AddProcessMapDialogSubmit").set('disabled', false);
-                } else if ((id === '') || (files.length < 1))
-                    registry.byId(this.id + "AddProcessMapDialogSubmit").set('disabled', true);
-                else
-                    registry.byId(this.id + "AddProcessMapDialogSubmit").set('disabled', false);
             },
 
             _onAddPackageMapSubmit: function () {
-                var target = this.addPackageMapTargetSelect.getValue();
-                var id = registry.byId(this.id + "AddProcessMapId").get('value');
-                var process = this.addPackageMapProcessSelect.getValue();
-                var daliIp = registry.byId(this.id + "AddProcessMapDaliIP").get('value');
-                var activate = registry.byId(this.id + "AddProcessMapActivate").get('checked');
-                var overwrite = registry.byId(this.id + "AddProcessMapOverWrite").get('checked');
-                if ((id === '') || (target === ''))
-                    return false;
-                if ((process === '') || (process === this.i18n.ANY))
-                    process = '*';
+                var context = this;
 
-                var action = "/WsPackageProcess/AddPackage?upload_&PackageMap=" + id + "&Target=" + target;
-                if (process !== '')
-                    action += "&Process=" + process;
-                if (daliIp !== '')
-                    action += "&DaliIp=" + daliIp;
-                if (activate)
-                    action += "&Activate=1";
-                else
-                    action += "&Activate=0";
-                if (overwrite)
-                    action += "&OverWrite=1";
-                else
-                    action += "&OverWrite=0";
-                var theForm = registry.byId(this.id + "AddProcessMapForm");
-                if (theForm === undefined)
-                    return false;
-                theForm.set('action', action);
-                return true;
+                if (this.addPackageMapForm.validate()) {
+                    WsPackageMaps.AddPackage({
+                        request: {
+                            Info: this.addPackageMapContent.value,
+                            PackageMap: this.addProcessMapId.get('value'),
+                            Process: this.addPackageProcessSelect.get('value'),
+                            Target: this.addPackageTargetSelect.get('value'),
+                            Activate: this.addProcessMapActivate.get('checked'),
+                            Overwrite: this.addProcessMapOverWrite.get('checked'),
+                            DaliIp: this.addProcessMapDaliIp.get('value')
+                        }
+                    }).then(function(response){
+                        if (lang.exists("AddPackageResponse.status", response)) {
+                            if (response.AddPackageResponse.status.Code === 0) {
+                                context.refreshGrid();
+                                context._onCancel();
+                            }
+                        } else {
+                            this._onCancel();
+                            this.showErrors(response.Exceptions.Exception[0].Message);
+                        }
+                    });
+                }
             },
 
             _onDelete: function (event) {
@@ -206,6 +176,8 @@ define([
                                 if (response.DeletePackageResponse.status.Code === 0) {
                                     context.refreshGrid();
                                 }
+                            } else {
+                                context.showErrors(response.Exceptions.Exception[0].Message);
                             }
                         });
                     });
@@ -257,16 +229,6 @@ define([
                 });
             },
 
-            addProcessSelections: function (processSelect, processes, processData) {
-                for (var i = 0; i < processData.length; ++i) {
-                    var process = processData[i];
-                    if ((processes != null) && (processes.indexOf(process) !== -1))
-                        continue;
-                    processes.push(process);
-                    processSelect.options.push({ label: process, value: process });
-                }
-            },
-
             init: function (params) {
                 var context = this;
                 if (this.inherited(arguments))
@@ -277,14 +239,13 @@ define([
                 this.targetSelect.init({
                     GetPackageMapTargets: true
                 });
-                
                 this.processSelect.init({
                     GetPackageMapProcesses: true
                 });
 
                 this.processSelectFilter.init({
                     GetPackageMapProcessFilter: true
-                })
+                });
 
                 this.addPackageTargetSelect.init({
                     GetPackageMapTargets: true
@@ -293,8 +254,6 @@ define([
                 this.addPackageProcessSelect.init({
                     GetPackageMapProcesses: true
                 });
-
-                this.initPackagesGrid();
 
                 this.filter.on("clear", function (evt) {
                     context._onFilterType();
@@ -305,6 +264,7 @@ define([
                     context.refreshHRef();
                     context.refreshGrid();
                 });
+                this.initPackagesGrid();
             },
 
             initTab: function () {
@@ -411,11 +371,8 @@ define([
             },
 
             getFilter: function () {
-                return {
-                    Target: this.targetSelect.getValue(),
-                    Process: this.processSelect.getValue(),
-                    ProcessFilter: this.processSelectFilter.getValue()
-                };
+                var retVal = this.filter.toObject();
+                return retVal;
             },
 
             refreshGrid: function (clearSelection) {
@@ -432,23 +389,6 @@ define([
                 registry.byId(this.id + "Delete").set("disabled", !hasSelection);
                 registry.byId(this.id + "Activate").set("disabled", selection.length !== 1);
                 registry.byId(this.id + "Deactivate").set("disabled", selection.length !== 1);
-            },
-
-            packageMapDeleted: function (tabId) {
-                if (this.tabMap[tabId] == null)
-                    return;
-                this.tabContainer.removeChild(this.tabMap[tabId]);
-                this.tabMap[tabId].destroyRecursive();
-                delete this.tabMap[tabId];
-
-                this.tabContainer.selectChild(this.packagesTab);
-                this.packagesGrid.rowSelectCell.toggleAllSelection(false);
-                this.refreshGrid();
-            },
-
-            addPackageMapCallback: function (event) {
-                this.addPackageMapDialog.onCancel();
-                this.refreshGrid();
             },
 
             ensurePane: function (id, params) {

--- a/esp/src/eclwatch/TargetSelectClass.js
+++ b/esp/src/eclwatch/TargetSelectClass.js
@@ -92,10 +92,16 @@ define([
                 } else if (params.DFUSprayQueues === true) {
                     this.loadSprayQueues();
                 } else if (params.GetPackageMapTargets === true) {
+                    this.defaultValue = "*";
+                    this.set("value", "*");
                     this.loadGetPackageMapTargets();
                 } else if (params.GetPackageMapProcesses === true) {
+                    this.defaultValue = "*";
+                    this.set("value", "*");
                     this.loadGetPackageMapProcesses();
                 } else if (params.GetPackageMapProcessFilter === true) {
+                    this.defaultValue = "*";
+                    this.set("value", "*");
                     this.loadGetPackageMapProcessFilter();
                 } else if (params.DropZoneMachines === true) {
                     this.defaultValue = "";

--- a/esp/src/eclwatch/templates/PackageMapQueryWidget.html
+++ b/esp/src/eclwatch/templates/PackageMapQueryWidget.html
@@ -22,50 +22,29 @@
                 <div id="${id}PackageMapsGridCP" style="border:0px; padding: 0px; border-color:none" data-dojo-props="region: 'center'" data-dojo-type="dijit.layout.ContentPane">
                     <div id="${id}PackagesGrid"></div>
                 </div>
-                <div id="${id}AddProcessMapDialog" data-dojo-type="dijit.Dialog" data-dojo-id="${id}AddProcessMapDialog" title="${i18n.AddProcessMap}">
-                    <div data-dojo-type="dijit.form.Form" id="${id}AddProcessMapForm" data-dojo-id="${id}AddProcessMapForm" method="post" enctype="multipart/form-data" >
-                       <table id="${id}AddProcessMapTable" class="dijitDialogPaneContentArea">
-                            <tr>
-                                <td><input id="${id}AddProcessMapFileUploader" name="${id}AddProcessMapFileUploader" multiple="false" type="file" data-dojo-type="dojox.form.Uploader" data-dojo-attach-event="onChange:_onCheckAddProcessMapInput" label="${i18n.SelectPackageFile}"/></td>
-                            </tr>
-                            <tr>
-                                <td><div id="${id}AddProcessMapFileToUpload" data-dojo-type="dojox.form.uploader.FileList" data-dojo-props='uploaderId:"${id}AddProcessMapFileUploader"'>
-                                </div></td>
-                            </tr>
-                            <tr>
-                                <td><label for="${id}AddProcessMapId" style="width: 15em;">${i18n.ID}:</label></td>
-                                <td><input name="${id}AddProcessMapId" id="${id}AddProcessMapId" data-dojo-attach-event="onKeyUp:_onAddProcessMapIdKeyUp" data-dojo-props="trim: true" data-dojo-type="dijit.form.TextBox"</td>
-                            </tr>
-                            <tr>
-                                <td><label for="${id}AddProcessMapTargetSelect">${i18n.Target}:</label></td>
-                                <td><div id="${id}AddProcessMapTargetSelect" name="${id}AddProcessMapTargetSelect" data-dojo-type="TargetSelectWidget">
-                                </div></td>
-                            </tr>
-                            <tr>
-                                <td><label for="${id}AddProcessMapProcessSelect">${i18n.ProcessFilter}:</label></td>
-                                <td>
-                                    <div id="${id}AddProcessMapProcessSelect" name="${id}AddProcessMapProcessSelect" data-dojo-type="TargetSelectWidget">
-                                    </div>
-                                </td>
-                            </tr>
-                            <tr>
-                                <td><label for="${id}AddProcessMapDaliIP">${i18n.RemoteDaliIP}:</label></td>
-                                <td><input name="${id}AddProcessMapDaliIP" id="${id}AddProcessMapDaliIP" data-dojo-props="trim: true" data-dojo-type="dijit.form.TextBox"></td>
-                            </tr>
-                            <tr>
-                                <td><label for="${id}AddProcessMapActivate">${i18n.Activate}:</label></td>
-                                <td><input id="${id}AddProcessMapActivate" name="${id}AddProcessMapActivate" data-dojo-type="dijit.form.CheckBox" value="" checked/></td>
-                            </tr>
-                            <tr>
-                                <td><label for="${id}AddProcessMapOverWrite">${i18n.Overwrite}:</label></td>
-                                <td><input id="${id}AddProcessMapOverWrite" name="${id}AddProcessMapOverWrite" data-dojo-type="dijit.form.CheckBox" value=""/></td>
-                            </tr>
-                            <tr>
-                                <td/>
-                                <td><input id="${id}AddProcessMapDialogSubmit" type="submit" label="${i18n.Submit}" data-dojo-type="dijit.form.Button"/>
-                                <input id="${id}AddProcessMapDialogClose" type="button" label="${i18n.Close}" data-dojo-type="dijit.form.Button"/></td>
-                            </tr>
-                        </table>
+                <div id="${id}AddProcessMapDialog" data-dojo-type="dijit.Dialog" title="${i18n.AddProcessMap}" style="width:700px; overflow-y: auto;">
+                    <div id="${id}AddProcessMapForm" data-dojo-type="dijit.form.Form" style="width:675px">
+                        <legend>Add Package Contents</legend>
+                        <div data-dojo-type="dijit.Fieldset">
+                            <div data-dojo-props="cols:1" data-dojo-type="hpcc.TableContainer">
+                                <textarea id="AddPackagemapContent" name="Info" rows="30" cols="50" style="width: 98%;"></textarea>
+                            </div>
+                        </div>
+                        <div data-dojo-type="dijit.Fieldset">
+                            <legend>Package Details</legend>
+                            <div data-dojo-props="cols:2" data-dojo-type="hpcc.TableContainer">
+                                <input id="${id}AddProcessMapId" name="PackageMap" title="${i18n.ID}:" colspan="2"  required="true" data-dojo-props="trim: true" data-dojo-type="dijit.form.ValidationTextBox" />
+                                <input id="${id}AddProcessMapTargetSelect" name="Target"  title="${i18n.Target}:" colspan="2" data-dojo-type="TargetSelectWidget" />
+                                <input id="${id}AddProcessMapProcessSelect" name="Process" title="${i18n.ProcessFilter}:" colspan="2" data-dojo-type="TargetSelectWidget" />
+                                <input id="${id}AddProcessMapDaliIP" name="DaliIp" title="${i18n.RemoteDaliIP}:" data-dojo-props="trim: true" colspan="2" data-dojo-type="dijit.form.TextBox" />
+                                <input id="${id}AddProcessMapActivate" name="Activate" title="${i18n.Activate}:" colspan="2" data-dojo-type="dijit.form.CheckBox" checked/>
+                                <input id="${id}AddProcessMapOverWrite" name="OverWrite" title="${i18n.Overwrite}:" colspan="2" data-dojo-type="dijit.form.CheckBox" />
+                            </div>
+                        </div>
+                        <div class="dijitDialogPaneActionBar">
+                            <button id="${id}AddProcessMapDialogClose" class="bottomFormButtons" data-dojo-attach-event="onClick:_onCancel" data-dojo-type="dijit.form.Button">${i18n.Cancel}</button>
+                            <button id="${id}AddProcessMapDialogSubmit" class="bottomFormButtons" data-dojo-attach-event="onClick:_onAddPackageMapSubmit" data-dojo-type="dijit.form.Button">${i18n.Submit}</button>
+                        </div>
                     </div>
                 </div>
             </div>

--- a/esp/src/src/WsPackageMaps.ts
+++ b/esp/src/src/WsPackageMaps.ts
@@ -46,6 +46,10 @@ export function AddPartToPackageMap(params) {
     return ESPRequest.send("WsPackageProcess", "AddPartToPackageMap", params);
 }
 
+export function AddPackage(params) {
+    return ESPRequest.send("WsPackageProcess", "AddPackage", params);
+}
+
 export function GetPartFromPackageMap(params) {
     return ESPRequest.send("WsPackageProcess", "GetPartFromPackageMap", params);
 }

